File: chroma-league/src/test/java/com/bonepl/chromaleague/hud/parts/resource/RenektonFuryBarTest.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.IntSteps;
import com.bonepl.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class RenektonFuryBarTest extends AbstractResourceTest {

    @Test
    void testRenektonFuryBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final RenektonFuryBar vladimirBloodPoolBar = new RenektonFuryBar();

        new AnimationTester()
                .withBeforeIterationAction(i -> {
                    if (i >= 20 && i <= 30) {
                        mockResource(100, 100);
                    } else {
                        mockResource(intSteps.nextInt(), 100);
                    }
                })
                .testAnimation(vladimirBloodPoolBar, 50);

    }
}

File: test/hud/parts/resource/RenektonFuryBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {RenektonFuryBar} from '../../../../src/hud/parts/resource/RenektonFuryBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for Renekton fury resource bar on actual hardware. */
describe('RenektonFury Integration', () => {
    setupResourceTest();

    test('testRenektonFuryBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const renektonFuryBar = new RenektonFuryBar();

        await new AnimationTester()
            .withBeforeIterationAction(i => {
                if (i >= 20 && i <= 30) {
                    mockResource(100, 100);
                } else {
                    mockResource(intSteps.nextInt(), 100);
                }
            })
            .testAnimation(renektonFuryBar, 50);
    }, 30000);
});
